### PR TITLE
Feature/split network from vpc

### PR DESCRIPTION
**Pull request by JanDoerrenhaus**

### CODE REVIEW

File: src/main/template/presto.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@ AWSTemplateFormatVersion: 2010-09-09
 Description: This template deploys a Presto cluster connected to a cassandra database.
 
 Parameters:
-  vpcStackName:
-    Description: The name of an existing VPC stack in which Presto will be deployed.
+  networkStackName:
+    Description: The name of an existing network stack in which a Cassandra cluster will be created.
```
Comment (schroenser) [inline src/main/template/presto.yaml]: Replace "a Cassandra cluster" with "Presto".

File: src/main/template/private-load-balancer.yaml [hunk 2]
```diff
@@ -2,8 +2,8 @@ AWSTemplateFormatVersion: 2010-09-09
 Description: This template deploys an application load balancer that is not reachable from the internet.
 
 Parameters:
-  vpcStackName:
-    Description: The name of an existing VPC stack in which the load balancer will be deployed.
+  networkStackName:
+    Description: The name of an existing network stack in which a Cassandra cluster will be created.
```
Comment (schroenser) [inline src/main/template/private-load-balancer.yaml]: Replace "a Cassandra cluster" with "the load balancer".

File: src/main/template/public-load-balancer.yaml [hunk 3]
```diff
@@ -2,8 +2,8 @@ AWSTemplateFormatVersion: 2010-09-09
 Description: This template deploys an application load balancer that is reachable from the internet.
 
 Parameters:
-  vpcStackName:
-    Description: The name of an existing VPC stack in which the load balancer will be deployed.
+  networkStackName:
+    Description: The name of an existing network stack in which a Cassandra cluster will be created.
```
Comment (schroenser) [inline src/main/template/public-load-balancer.yaml]: Replace "a Cassandra cluster" with "the load balancer".

File: src/main/template/redis.yaml [hunk 4]
```diff
@@ -2,8 +2,8 @@ AWSTemplateFormatVersion: 2010-09-09
 Description: This template deploys an ElastiCache managed Redis.
 
 Parameters:
-  vpcStackName:
-    Description: The name of an existing VPC stack in which Redis will be deployed.
+  networkStackName:
+    Description: The name of an existing network stack in which a Cassandra cluster will be created.
```
Comment (schroenser) [inline src/main/template/redis.yaml]: Replace "a Cassandra cluster" with "Redis".

**Review by schroenser — CHANGES_REQUESTED** [review]

**Comment by schroenser** [comment]
Do we have any strategy for merging this pull request?
The changes made here require that we completely re-create all of our stacks, including data migration in Cassandra, PostgreSQL and MySQL.

However, if me merge this now and want to create a new stack (let's say for cumulus), it won't be compatible to our current stacks.
We could release a `1.0` before merging, marking this as a breaking change to the "API" of the stacks.

**Review by schroenser — CHANGES_REQUESTED** [review]
Please also, update the image "VPC Architecture" shown in the README.adoc.

I think it should now show the VPC containing one (or more?) "public-private-subnets-3az" with the rest of the current contents of the image as well as an additional empty "some-other-subnet-structure" as an indication for what can be achieved with the template structure.

**Comment by JanDoerrenhaus** [comment]
> 
> 
> Do we have any strategy for merging this pull request?
> The changes made here require that we completely re-create all of our stacks, including data migration in Cassandra, PostgreSQL and MySQL.
> 
> However, if me merge this now and want to create a new stack (let's say for cumulus), it won't be compatible to our current stacks.
> We could release a `1.0` before merging, marking this as a breaking change to the "API" of the stacks.

The good thing about having version control is that older versions do not cease to exist. Cumulus will mostly use existing infrastructure anyway. Only live Cumulus will create its own Cassandra Cluster, and for that, we can easily keep using 0.3.

**Comment by JanDoerrenhaus** [comment]
> 
> 
> Please also, update the image "VPC Architecture" shown in the README.adoc.
> 
> I think it should now show the VPC containing one (or more?) "public-private-subnets-3az" with the rest of the current contents of the image as well as an additional empty "some-other-subnet-structure" as an indication for what can be achieved with the template structure.

I would _really_ rather not. First of all, the image has neither become wrong nor misleading. The standard use case is still to have just one network in your VPC, which would look exactly like the image suggests. Having multiple, differently configured networks in your VPC is a non-standard use case. Second, the template designer does not allow you to draw freeform, only to visualize existing resources, so "some-other-subnet-structure" is quite impossible. Third, that designer is a pain to work with, and even just recreating this image will take hours.

**Comment by JanDoerrenhaus** [comment]
> 
> 
> > Please also, update the image "VPC Architecture" shown in the README.adoc.
> > I think it should now show the VPC containing one (or more?) "public-private-subnets-3az" with the rest of the current contents of the image as well as an additional empty "some-other-subnet-structure" as an indication for what can be achieved with the template structure.
> 
> I would _really_ rather not. First of all, the image has neither become wrong nor misleading. The standard use case is still to have just one network in your VPC, which would look exactly like the image suggests. Having multiple, differently configured networks in your VPC is a non-standard use case. Second, the template designer does not allow you to draw freeform, only to visualize existing resources, so "some-other-subnet-structure" is quite impossible. Third, that designer is a pain to work with, and even just recreating this image will take hours.

As a concession, I have added some explanatory text.

**Review by schroenser — APPROVED** [review]

**Comment by schroenser** [comment]
fyi @bannmann

**Pull request merged**